modules/geochemistry/include/utils/GeochemistryUnitConverter.h

```diff
@@ -0,0 +1,101 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+#include "MooseTypes.h"
+#include "PertinentGeochemicalSystem.h"
+
+/**
+ * Utilities to convert to and from mole units
+ */
+namespace GeochemistryUnitConverter
+{
+enum class GeochemistryUnits
+{
+  DIMENSIONLESS,
+  MOLES,
+  MOLAL,
+  KG,
+  G,
+  MG,
+  UG,
+  KG_PER_KG_SOLVENT,
+  G_PER_KG_SOLVENT,
+  MG_PER_KG_SOLVENT,
+  UG_PER_KG_SOLVENT,
+  CM3
+};
+
+/**
+ * Calculates the amount of "unit"s of species_name in 1 mole, OR in 1 molal, whichever is
+ * appropriate.  Example: if moles = 2.0 and unit = CM3 and species_name = Halite, this function
+ * will return the number of cm^3 in 1 mole of Halite.  Example: if moles = 2.0 and unit = MG_PER_KG
+ * and species_name = Halite, this function will return the number of mg/kg_solvent_water in a 2.0
+ * molal solution of Halite.
+ * @param moles the amount of moles of species_name
+ * @param unit the unit to convert to.  If unit is DIMENSIONLESS,
+ * MOLES or MOLAL, this returns quantity.  If unit is KG, G, MG, UG or CM3, then "moles" is the mole
+ * number.  If unit is KG_PER_KG_SOLVENT, G_PER_KG_SOLVENT, MG_PER_KG_SOLVENT, UG_PER_KG_SOLVENT,
+ * then "moles" is assumed to be the molality.
+ * @param species_name the name of the substance - this should appear in mgd
+ * @param mgd the database corresponding to the model, which should contain information about
+ * species_name
+ */
+Real fromMoles(Real moles,
+               const GeochemistryUnits & unit,
```

Comment:
```suggestion
               const GeochemistryUnits unit,
```

FYI with something like this enum that is an int, having this as a reference could be slower! There's a few others below.

Reply:
Aha, i get it, thanks.